workflow/lib/shared/segment_cellpose.py

```diff
@@ -148,15 +155,73 @@ def prepare_cellpose(data, dapi_index, cyto_index, logscale=True, log_kwargs=dic
     return np.array([red, green, blue])
 
 
+def estimate_diameters(
+    data,
+    dapi_index,
+    cyto_index,
+    channels=[2, 3],  # Default channels for cell estimation  
+    cyto_model="cyto3",
+    cellpose_kwargs=dict(
+        flow_threshold=0.4,    
+        cellprob_threshold=0
+    ),
+    gpu=False,
+    logscale=True,
+):
+    """Estimate optimal cell diameter using Cellpose's diameter estimation.
+    
+    Args:
+        data (numpy.ndarray): Multichannel image data
+        dapi_index (int): Index of DAPI channel
+        cyto_index (int): Index of cytoplasmic channel
+        channels (list): Channel indices for diameter estimation [cytoplasm, nuclei]
+        cyto_model (str): Cellpose model type to use
+        gpu (bool): Whether to use GPU
+        logscale (bool): Whether to apply log scaling to image
+        
+    Returns:
+        tuple: Estimated diameters for (nuclei, cells)
+    """
+
+    # Prepare RGB image
+    log_kwargs = cellpose_kwargs.pop(
+        "log_kwargs", dict()
+    )  # Extract log_kwargs from cellpose_kwargs
+    rgb = prepare_cellpose(
+        data, dapi_index, cyto_index, logscale, log_kwargs=log_kwargs
+    )
+    
+    # Find optimal nuclei diameter
+    print("Estimating nuclei diameters...")
+    model_nuclei = Cellpose(model_type="nuclei", gpu=gpu)
+    diam_nuclear, _ = model_nuclei.sz.eval(rgb, channels=[3, 0])
+    diam_nuclear = np.maximum(5.0, diam_nuclear)
+    diam_nuclear = float(diam_nuclear)
+    print(f"Estimated nuclear diameter: {diam_nuclear:.1f} pixels")
+
+    # Find optimal cell diameter
+    print("Estimating cell diameters...")
+    model_cyto = Cellpose(model_type=cyto_model, gpu=gpu)
+    diam_cell, _ = model_cyto.sz.eval(rgb, channels=channels)
+    diam_cell = np.maximum(5.0, diam_cell)  
+    diam_cell = float(diam_cell)
+    print(f"Estimated cell diameter: {diam_cell:.1f} pixels")
+        
+    return diam_nuclear, diam_cell
+
+
 def segment_cellpose_rgb(
     rgb,
     nuclei_diameter,
     cell_diameter,
-    gpu=False,
     cyto_model="cyto3",
     reconcile="consensus",
     remove_edges=True,
     return_counts=False,
+    flow_threshold=0.4,
+    cellprob_threshold=0,
+    gpu=False,
+    **kwargs
```

Comment:
Another issue that ruff was throwing is that there are missing param definitions in this docstring. Sorry I haven't gotten to the GitHub action for linting yet :(